File: loadJSON.js
import {State, Attack, AttackDB} from "./makeChain.js";
import fs from 'fs';

function loadAttacks(attackData){
    // For Local Use
    /*
    const jsonData = fs.readFileSync(filename, 'utf8');
    let attackData;
    try {
        attackData = JSON.parse(jsonData);
    } catch (error) {
        console.error(error);
    }*/

    function createAttack(data) {
        return data.map(attack => {
        const { name, initState, endState } = attack;
        return new Attack(name, new State(initState), new State(endState));
        });
    }
    return createAttack(attackData);
}

function loadState(filename){
    const jsonData = fs.readFileSync(filename, 'utf8');

    let state;
    try {
        state = JSON.parse(jsonData);
    } catch (error) {
        console.error(error);
    }

    return new State(state);
}

// Just a logging function
function logAttacks(attakcs){
    attacks.forEach((attack, index) => {
        console.log(`Attack ${index + 1}:`);
        console.log(`Name: ${attack.name}`);
        console.log(`Init State: ${JSON.stringify(attack.initState)}`);
        console.log(`End State: ${JSON.stringify(attack.endState)}`);
        console.log('\n');
    })
}


export {loadAttacks, loadState, logAttacks};

File: backend/helpers/loadJSON.py
import json
import logging
from helpers.makeChain import State, Attack, AttackDB

logging.basicConfig(level=logging.INFO)


# This function takes a list of dictionaries, where each dictionary represents an attack.
# It creates an Attack object for each dictionary and returns a list of these Attack objects.
def create_attack(data):
    return [
        Attack(
            attack["name"],
            State(attack["initState"]),
            State(attack["endState"]),
            set(attack["info_required"]),
            set(attack["info_gained"]),
        )
        for attack in data
    ]


# This function takes a filename as input, opens the file, and loads the JSON data from the file.
# It then passes this data to the create_attack function to create a list of Attack objects, and returns this list.
def load_attacks(filename):
    with open(filename, "r") as file:
        data = json.load(file)
    return create_attack(data)


# This function takes a filename as input, opens the file, and loads the JSON data from the file.
# It then creates a State object from this data and returns it.
def load_state(filename):
    with open(filename, "r") as file:
        data = json.load(file)
    return State(data)
